runtime/battery-station/src/xcm_config/config.rs

```diff
@@ -0,0 +1,305 @@
+impl TakeRevenue for ToTreasury {
+    fn take_revenue(revenue: MultiAsset) {
+        use xcm_executor::traits::Convert;
+
+        if let MultiAsset { id: Concrete(location), fun: Fungible(amount) } = revenue.clone() {
+            if let Ok(asset_id) =
+                <AssetConvert as Convert<MultiLocation, CurrencyId>>::convert(location.clone())
+            {
+                let _ = AssetManager::deposit(asset_id, &ZeitgeistTreasuryAccount::get(), amount);
+            }
+        }
+    }
+}
+
+parameter_types! {
+    pub CheckAccount: AccountId = PolkadotXcm::check_account();
+    /// The amount of ZTG charged per second of execution (canonical multilocation).
+    pub ZtgPerSecondCanonical: (AssetId, u128) = (
+        MultiLocation::new(
+            0,
+            X1(general_key(battery_station::KEY)),
+        ).into(),
+        native_per_second(),
+    );
+    /// The amount of canonical ZTG charged per second of execution.
+    pub ZtgPerSecond: (AssetId, u128) = (
+        MultiLocation::new(
+            1,
+            X2(Junction::Parachain(ParachainInfo::parachain_id().into()), general_key(battery_station::KEY)),
+        ).into(),
+        native_per_second(),
+    );
+}
+
+/// Means for transacting assets on this chain.
+pub type MultiAssetTransactor = MultiCurrencyAdapter<
+    // All known Assets will be processed by the following MultiCurrency implementation.
+    AssetManager,
+    // Any unknown Assets will be processed by the following UnknownAsset implementation.
+    UnknownTokens,
+    // This means that this adapter should handle any token that `AssetConvert` can convert
+    // using AssetManager and UnknownTokens in all other cases.
+    IsNativeConcrete<CurrencyId, AssetConvert>,
+    // Our chain's account ID type (we can't get away without mentioning it explicitly).
+    AccountId,
+    // Convert an XCM `MultiLocation` into a local account id.
+    LocationToAccountId,
+    // The AssetId that corresponds to the native currency.
+    CurrencyId,
+    // Struct that provides functions to convert `Asset` <=> `MultiLocation`.
+    AssetConvert,
+    // In case of deposit failure, known assets will be placed in treasury.
+    DepositToAlternative<ZeitgeistTreasuryAccount, AssetManager, CurrencyId, AccountId, Balance>,
+>;
+
+/// AssetConvert
+/// This type implements conversions from our `Asset` type into `MultiLocation` and vice-versa.
+/// A currency locally is identified with a `Asset` variant but in the network it is identified
+/// in the form of a `MultiLocation`, in this case a pair (Para-Id, Currency-Id).
+pub struct AssetConvert;
+
+/// Convert our `Asset` type into its `MultiLocation` representation.
+/// Other chains need to know how this conversion takes place in order to
+/// handle it on their side.
+impl Convert<CurrencyId, Option<MultiLocation>> for AssetConvert {
+    fn convert(id: CurrencyId) -> Option<MultiLocation> {
+        match id {
+            Asset::Ztg => Some(MultiLocation::new(
```

Comment:
It is `CurrencyId::Ztg` on any chain. I think this is hard to misunderstand and having a unique name for each network would demand for serious refractoring. Value<->effort is not worth it. Especially since the UI can easily map it for the convenience of the user.
Instead of `CurrencyId::Ztg`, we could use `CurrencyId::Native` though. This would be a general purpose identifier for the native currency of a chain.